CMakeLists.txt

```diff
@@ -2,39 +2,51 @@ cmake_minimum_required(VERSION 3.20)
 
 project(libkmeans LANGUAGES Fortran)
 
-# Build directory
-set(CMAKE_Fortran_BIN_DIRECTORY ${PROJECT_BINARY_DIR}/bin)
-set(CMAKE_Fortran_LIB_DIRECTORY ${PROJECT_BINARY_DIR}/lib)
 set(CMAKE_Fortran_MODULE_DIRECTORY ${PROJECT_BINARY_DIR}/modules)
 
 # Compiler flags for GCC only
+# Note, flags "should" be specified in yaml presets, and passed to cmake as arguments
+# Use c-preprocessor. Note, probably better to set target property `Fortran_PREPROCESS` instead
 set(GCC_BASE 
-   -fopenmp   # OMP always enabled
-   -cpp       # Use c-preprocessor
+   -cpp       
    )
 
 set(GCC_DEBUG
      -O0                                          # 0 optimisation. Probably implicit
      -fbacktrace                                  # symbolic stack traceback
-   #   -ffpe-trap=invalid,zero,overflow,underflow   # control over floating-point exception
-   #   -finit-real=nan                              # All real scalars are initialised to NaN
      -fcheck=all                                  # Enable all run-time test of -fcheck: array-temps, bits, bounds, do, mem, pointer, recursion
      -Wall                                        # Enable all warnings
+ #   -ffpe-trap=invalid,zero,overflow,underflow   # control over floating-point exception
+ #   -finit-real=nan                              # All real scalars are initialised to NaN
     )
-
+                    
 set(GCC_RELEASE
     -O3
    )
 
-# Convert list to string
-list(JOIN GCC_BASE " " GCC_BASE)
-list(JOIN GCC_DEBUG " " GCC_DEBUG)
-list(JOIN GCC_RELEASE " " GCC_RELEASE)
+# When appending existing cmake flags (which one should not do), one converts list to string
+# list(JOIN GCC_BASE " " GCC_BASE)
+# set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} ${GCC_BASE}")  
+
+# When using an interface target to assign compiler flags (or options), one requires a semicolon-separated string
+# which is == strings appended to a list
+# Convert the space-separated string into a semicolon-separated string
+string(REPLACE " " ";" GCC_BASE "${GCC_BASE}")
+string(REPLACE " " ";" GCC_DEBUG "${GCC_DEBUG}")
+string(REPLACE " " ";" GCC_RELEASE "${GCC_RELEASE}")
```

Comment:
What are you doing here? `set(var item_1 item_2 ....)` makes `var` a list already.